pkg/api/entries.go

```diff
@@ -91,9 +92,15 @@ func CreateLogEntryHandler(params entries.CreateLogEntryParams) middleware.Respo
 	}
 
 	resp := tc.addLeaf(leaf)
-	switch resp.status {
-	case codes.OK:
-	case codes.AlreadyExists, codes.FailedPrecondition:
+	//this represents overall GRPC response state (not the results of insertion into the log)
+	if resp.status != codes.OK {
+		return handleRekorAPIError(params, http.StatusInternalServerError, fmt.Errorf("grpc error: %w", resp.err), trillianUnexpectedResult)
+	}
+
+	//this represents the results of inserting the proposed leaf into the log
+	switch resp.getAddResult.QueuedLeaf.Status.Code {
```

Comment:
tiny nit: I think you can do something like `switch int32(resp.getAddResult.QueuedLeaf.Status.Code)` to save the repeated casting below